analyzers/tests/SonarAnalyzer.UnitTest/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.cs

```diff
@@ -110,31 +110,38 @@ public void UsedAfterBranch_LiveOut()
             context.Validate(context.Cfg.ExitBlock);
         }
 
-        [TestMethod]
-        public void Captured_NotLiveIn_NotLiveOut()
-        {
-            /*       Binary
+        [DataTestMethod]
+        [DataRow("Capturing(x => field + variable + intParameter);", DisplayName = "SimpleLambda")]
+        [DataRow("Capturing((x) => field + variable + intParameter);", DisplayName = "ParenthesizedLambda")]
+        [DataRow("Capturing(x => { System.Func<int> xxx = () => field + variable + intParameter; return xxx();});", DisplayName = "NestedLambda")]
+        [DataRow("VoidDelegate d = delegate { return field + variable + intParameter;};", DisplayName = "AnonymousMethod")]
+        [DataRow("var items = from xxx in new int[] { 42, 100 } where xxx > field + variable + intParameter select xxx;", DisplayName = "Query")]
+        public void Captured_NotLiveIn_NotLiveOut(string capturingStatement)
+        {
+            /*       Entry
+             *         |
+             *       Block 1
              *       /   \
-             *    Jump   Simple
-             *   return  Method()
+             *      |   Block 2
+             *      |   Method()
              *       \   /
              *        Exit
              */
-            var code = @"
-Capturing(() => intParameter);
+            var code = @$"
+var variable = 42;
+{capturingStatement}
 if (boolParameter)
     return;
-Method(intParameter);";
+Method(field, variable, intParameter);";
+            capturingStatement.Should().Contain("field + variable + intParameter");
             var context = new Context(code);
-            TmpNotImplemented();
-            //var binary = context.Cfg.EntryBlock;
-            //var jump = context.Block<JumpBlock>();
-            //var simple = context.Block<SimpleBlock>();
-            //var exit = context.Cfg.ExitBlock;
-            //context.Validate(binary, new Captured("intParameter"), new LiveIn("boolParameter"));
-            //context.Validate(jump, new Captured("intParameter"));
-            //context.Validate(simple, new Captured("intParameter"));
-            //context.Validate(exit, new Captured("intParameter"));
+            var expectedCaptured = capturingStatement.Contains("xxx")
```

Comment:
Do we really need it? Can we just split this into 2 tests?

Reply:
I want every other test to assert that the captures are actually empty with a very easy scaffolding. 

The logic is a little mixed here because captures do not depend on a block.